templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -139,6 +140,7 @@ contents:
                     # re-run the lookup of the hostname now that we know we have DNS
                     # servers configured correctly in resolv.conf.
                     nmcli general reload dns-rc
+                    touch "${KNICONFDONEPATH}"
```

Comment:
This unifies both flows so that KNICONFDONEPATH be created in the end - and kubelet dropin logic would be simplified just to check if this file exists (regardless of the systemd-resolved presence)

Reply:
I see. I didn't consider the dropin logic. The name is now a bit weird for what youre trying to accomplish.